crates/hir_expand/src/lib.rs

```diff
@@ -285,14 +300,17 @@ impl MacroCallKind {
             MacroCallKind::FnLike { ast_id, .. } => {
                 Some(ast_id.to_node(db).token_tree()?.syntax().clone())
             }
-            MacroCallKind::Derive { ast_id, .. } => Some(ast_id.to_node(db).syntax().clone()),
+            MacroCallKind::Derive { ast_id, .. } | MacroCallKind::Attr { ast_id, .. } => {
+                Some(ast_id.to_node(db).syntax().clone())
+            }
         }
     }
 
     fn fragment_kind(&self) -> FragmentKind {
         match self {
             MacroCallKind::FnLike { fragment, .. } => *fragment,
             MacroCallKind::Derive { .. } => FragmentKind::Items,
+            MacroCallKind::Attr { .. } => FragmentKind::Items, // is this always correct?
```

Comment:
No, you can apply an attribute to a statement or expression. It is unstable for expressions, but stable for statements.

```rust
#![feature(stmt_expr_attributes)]
fn main() {
    let _ = #[allow(dead_code)] { panic!(); 0 };
}
```

```rust
fn main() {
    #[allow(unused_variables)]
    let foo = 0;
}
```

Reply:
this is also still unstable for proc macro attributes:

![screenshot-2021-06-03-16:21:18](https://user-images.githubusercontent.com/1786438/120660604-cafadf80-c487-11eb-9b57-1e5be7a71d40.png)

Reply:
I see.